src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java

```diff
@@ -292,6 +322,34 @@ public void setCanceled(boolean canceled)
                 useItem = DENY;
             }
         }
+
+        public static enum Action {
+        	/**
+        	 * When the player first left clicks a block
+        	 */
+        	START,
+        	/**
+        	 * When the player stops left clicking a block by completely breaking it
+        	 */
+        	STOP,
+        	/**
+        	 * When the player stops left clicking a block by releasing the button, or no longer targeting the same block
```

Comment:
Probably should say 'before it breaks' unless there is a scenario where both stop and abort can be called.